src/System.IdentityModel.Tokens/SymmetricSecurityKey.cs

```diff
@@ -48,7 +48,16 @@ public override int KeySize
 
         public override SignatureProvider GetSignatureProvider(string algorithm, bool forSigning)
         {
-            return null;
+            var factory = this.SignatureProviderFactory ?? SignatureProviderFactory.Default;
+
+            if (forSigning)
+            {
+                return factory.CreateForSigning(this, algorithm);
```

Comment:
If we do close on supporting SHA1 for verifying, we will need to modify factory.CreateForSigning to fault if SHA1 is detected.